osgtest/tests/test_450_xrootd.py

```diff
@@ -11,188 +11,265 @@
+            os.chmod(file_to_download, 0o644)
 
+    @xrootd_record_failure
     def test_03a_xrdcp_upload_gsi_authenticated(self):
-        self.skip_ok_unless("GSI" in core.config['xrootd.security'], "not using GSI")
-        self.skip_bad_unless(core.state['proxy.valid'], "no valid proxy")
-        try:
-            self.skip_bad_unless_running(core.config['xrootd_service'])
-            xrootd_url = self.xrootd_url(TestXrootd.user_copied_file)
-            command = ('xrdcp', '--debug', '3', TestXrootd.__data_path, xrootd_url)
-            core.check_system(command, "Authenticated xrdcp upload to private dir", user=True)
-            self.assert_(os.path.exists(TestXrootd.user_copied_file), "Uploaded file missing")
-        except AssertionError:
-            core.state['xrootd.had-failures'] = True
-            raise
+        self.skip_unless_security("GSI")
+        command = ('xrdcp', '--debug', '2', TestXrootd.__data_path, xroot_url(TestXrootd.user_copied_file_gsi))
+        with core.no_bearer_token(core.options.username):
+            core.check_system(command, "xrdcp upload to user dir with GSI auth", user=True)
+        self.assert_(os.path.exists(TestXrootd.user_copied_file_gsi), "Uploaded file missing")
 
+    @xrootd_record_failure
     def test_03b_xrdcp_upload_scitoken_authenticated(self):
-        self.skip_ok_unless("SCITOKENS" in core.config['xrootd.security'], "not using scitokens")
-        token_contents = core.state['token.xrootd_contents']
-        self.skip_bad_unless(token_contents, "xrootd scitoken not found")
-        try:
-            self.skip_bad_unless_running(core.config['xrootd_service'])
-            bearer_token = token_contents if core.config['xrootd.ztn'] else None
-            with core.environ_context("BEARER_TOKEN", bearer_token):
-                    xrootd_url = self.xrootd_url(TestXrootd.user_copied_file)
-                    command = ('xrdcp', '--debug', '3', TestXrootd.__data_path, xrootd_url)
-                    core.check_system(command, "Authenticated xrdcp upload to private dir", user=True)
-                    self.assert_(os.path.exists(TestXrootd.user_copied_file), "Uploaded file missing")
-        except AssertionError:
-            core.state['xrootd.had-failures'] = True
-            raise
+        self.skip_unless_security("SCITOKENS")
+        xrootd_url = xroot_url(TestXrootd.user_copied_file_scitoken)
+        command = ('xrdcp', '--force', '--debug', '2', TestXrootd.__data_path, xrootd_url)
+        with core.no_x509(core.options.username):
+            with core.environ_context({"BEARER_TOKEN": core.state['token.xrootd_contents'], "BEARER_TOKEN_FILE": None}):
+                message = "xrdcp upload to user dir with scitoken in BEARER_TOKEN"
+                expected_exit = 0
+                if core.PackageVersion("xrootd-libs") <= "5.3.2":
+                    message += " (expected failure)"
+                    expected_exit = ERR_AUTH_FAIL
```

Comment:
```suggestion
                if core.PackageVersion("xrootd-libs") <= "5.3.2":
                    message += " (expected failure)"
                    expected_exit = ERR_AUTH_FAIL
```

A comment for why this is the case would be helpful

Reply:
I'll file a bug report and add a comment pointing to the issue.